ex02.exs

```diff
@@ -1,8 +1,31 @@
 
 defmodule Ex02 do
+  @name __MODULE__
+
+  def new_counter(value \\ 0) do
+    { :ok, counter } = Agent.start_link fn -> value end   #create new agent process with a value
+    counter                                               #and return the pid
+  end
+
+  def next_value(counter_pid) do
```

Comment:
There's a race condition in this code. Can you spot it?

We'll talk about this as a group in class.

### -3